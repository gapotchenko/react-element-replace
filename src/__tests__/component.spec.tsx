import * as React from 'react';
import { Replacer } from "..";
import { create } from 'react-test-renderer';

function TestComponent(props: {
    label?: string;
    children?: any;
}) {
    return (<div>
        {props.label} :{" "}
        <div style={{
            display: "inline-block",
            border: "1px solid black"
        }}>
            Label Me
      </div>
        <div>{props.children}</div>
    </div>);
}

test("component replace", () => {
    const component = create(
        <Replacer
            matchElement={TestComponent}
            replace={({ type: TestComponent, props }: any) => (
                <TestComponent {...props} label="The Label" />
            )}>
            <TestComponent>
                <TestComponent />
            </TestComponent>
        </Replacer>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
