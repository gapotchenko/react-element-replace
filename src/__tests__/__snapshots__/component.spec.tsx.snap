// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component replace 1`] = `
<div>
   :
   
  <div
    style={
      {
        "border": "1px solid black",
        "display": "inline-block",
      }
    }
  >
    Label Me
  </div>
  <div>
    <div>
       :
       
      <div
        style={
          {
            "border": "1px solid black",
            "display": "inline-block",
          }
        }
      >
        Label Me
      </div>
      <div />
    </div>
  </div>
</div>
`;
